### PR TITLE
drivers: flash: nordic qspi: finish DT API conversion

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Also convert bus node access to the new API.

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
use DT_INST form since driver is already doing that for other properties.

File: drivers/flash/nrf_qspi_nor.c [hunk 1]
```diff
@@ -26,6 +26,10 @@
 /* instance 0 flash size in bytes */
 #define INST_0_BYTES (DT_INST_PROP(0, size) / 8)
 
+/* for accessing devicetree properties of the bus node */
+#define QSPI_NODE DT_BUS(DT_DRV_INST(0))
+#define QSPI_PROP_AT(prop, idx) DT_PROP_BY_IDX(QSPI_NODE, prop, idx)
```
Comment (galak) [inline drivers/flash/nrf_qspi_nor.c:31]: Make this `DT_INST_PROP_BY_IDX(0, prop, idx)`

File: drivers/flash/nrf_qspi_nor.c [hunk 2]
```diff
@@ -363,12 +367,12 @@ static inline void qspi_fill_init_struct(nrfx_qspi_config_t *initstruct)
 	initstruct->xip_offset = 0;
 
 	/* Configure pins */
-	initstruct->pins.sck_pin = DT_NORDIC_NRF_QSPI_QSPI_0_SCK_PIN;
-	initstruct->pins.csn_pin = DT_NORDIC_NRF_QSPI_QSPI_0_CSN_PINS_0;
-	initstruct->pins.io0_pin = DT_NORDIC_NRF_QSPI_QSPI_0_IO_PINS_0;
-	initstruct->pins.io1_pin = DT_NORDIC_NRF_QSPI_QSPI_0_IO_PINS_1;
-	initstruct->pins.io2_pin = DT_NORDIC_NRF_QSPI_QSPI_0_IO_PINS_2;
-	initstruct->pins.io3_pin = DT_NORDIC_NRF_QSPI_QSPI_0_IO_PINS_3;
+	initstruct->pins.sck_pin = DT_PROP(QSPI_NODE, sck_pin);
```
Comment (galak) [inline drivers/flash/nrf_qspi_nor.c:370]: `DT_INST_PROP(0, sck_pin)`

File: drivers/flash/nrf_qspi_nor.c [hunk 3]
```diff
@@ -628,8 +632,7 @@ static int qspi_nor_configure(struct device *dev)
  */
 static int qspi_nor_init(struct device *dev)
 {
-	IRQ_CONNECT(DT_NORDIC_NRF_QSPI_QSPI_0_IRQ_0,
-		    DT_NORDIC_NRF_QSPI_QSPI_0_IRQ_0_PRIORITY,
+	IRQ_CONNECT(DT_IRQN(QSPI_NODE), DT_IRQ(QSPI_NODE, priority),
```
Comment (galak) [inline drivers/flash/nrf_qspi_nor.c:635]: DT_INST_IRQ...

**Comment by mbolivar-nordic** [comment]
> use DT_INST form since driver is already doing that for other properties.

No, that's not correct. DT_DRV_COMPAT is nordic_qspi_nor. QSPI_NODE is the bus node; its compatible is nordic,nrf-qspi. They're different nodes.

**Comment by galak** [comment]
> No, that's not correct. DT_DRV_COMPAT is nordic_qspi_nor. QSPI_NODE is the bus node; its compatible is nordic,nrf-qspi. They're different nodes.

Ah, missed that its the BUS node.

**Review by galak — APPROVED** [review]

**Review by kamlaz — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**